### PR TITLE
TeacherFeedback Refactor #1

### PR DESCRIPTION
**Pull request by maureensturgeon**
This is a refactor with no functional changes. It moves TeacherFeedback queries into teacher_feedback.rb and renames them to represent more clearly what the return values are.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/models/teacher_feedback.rb [hunk 1]
```diff
@@ -67,7 +67,17 @@ def self.get_student_level_feedback(student_id, level_id, teacher_id, script_id)
...
+  def self.get_latest_feedbacks_received(student_id, level_id, script_id)
+    where(
+      student_id: student_id,
+      level_id: level_id,
+      script_id: script_id
+    ).latest_per_teacher
+  end
+
+  # returns the latest feedback for each student on every level given by the teacher
+  def self.get_latest_feedbacks_given(student_ids, level_ids, teacher_id)
```
Comment (jamescodeorg) [inline dashboard/app/models/teacher_feedback.rb]: Should usage of this method be replaced with the method on L61?  (We should always be using both script_id and level_id, right?)

Comment (maureensturgeon) [reply dashboard/app/models/teacher_feedback.rb]: Oh! I'll add `script_id` to the query, but the difference between this and the one on line 61, is this one returns multiple feedbacks whereas the other one only returns one for one level and one student

Comment (jamescodeorg) [reply dashboard/app/models/teacher_feedback.rb]: Oh, is the difference that the one on 61 only returns for one student but this one returns for multiple students?  If that's the case, can we replace calls to 61 with this method (passing in an array with a single student id)?

Comment (maureensturgeon) [reply dashboard/app/models/teacher_feedback.rb]: This one takes a list of student_ids and level_ids, I'll see what I can do!

Comment (maureensturgeon) [reply dashboard/app/models/teacher_feedback.rb]: Updated!

File: apps/src/templates/feedback/StudentFeedbackNotification.jsx [hunk 2]
```diff
@@ -35,20 +35,20 @@ export default class StudentFeedbackNotification extends Component {
       numFeedbackLevels: this.state.numFeedbackLevels
     });
 
+    if (!this.state.numFeedbackLevels) {
+      return null;
```
Comment (cforkish) [inline apps/src/templates/feedback/StudentFeedbackNotification.jsx:39]: 👍

**Review by cforkish — APPROVED** [review]

**Review by jamescodeorg — APPROVED** [review]
Were there any changes to existing tests or new tests that should be added?

**Comment by maureensturgeon** [comment]
> Were there any changes to existing tests or new tests that should be added?

Going to leave as-is for now since there were no functionality changes. I'll be revisiting teacher_feedback.rb a lot over the next couple of weeks which will give me more opportunities to add to the test suite.

**Pull request merged**